consensus/src/ordering_state_computer.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{error::StateSyncError, state_replication::StateComputer};
+use anyhow::Result;
+use consensus_types::{block::Block, executed_block::ExecutedBlock};
+use diem_crypto::{HashValue, hash::ACCUMULATOR_PLACEHOLDER_HASH};
+use diem_infallible::Mutex;
+use diem_metrics::monitor;
+use diem_types::ledger_info::LedgerInfoWithSignatures;
+use executor_types::{Error as ExecutionError, StateComputeResult};
+use fail::fail_point;
+use std::{boxed::Box, sync::Arc};
+use channel::Sender;
+
+/// Basic communication with the Execution module;
+/// implements StateComputer traits.
```

Comment:
maybe update the comments to describe what this struct is supposed to do compared to the other one